test/common/TestBedChild.cpp

```diff
@@ -34,20 +38,30 @@
     if (ncclAsyncErr != ncclSuccess)                                  \
     {                                                                 \
       ERROR("Child process %d fails NCCL call %s with code %d\n", this->childId, msg, ncclAsyncErr);  \
-      return TEST_FAIL;                                               \
+      RESULT(TEST_FAIL, ##RESULT_ARGS);                               \
     }                                                                 \
-  }
+  } while (false)
+#define CHILD_NCCL_CALL_NON_BLOCKING(msg, localRank) CHILD_NCCL_CALL_NON_BLOCKING_BASE(msg, localRank, RETURN_RESULT)
 
 #define PIPE_READ(val) \
   if (read(childReadFd, &val, sizeof(val)) != sizeof(val)) return TEST_FAIL;
 
+#ifdef ENABLE_OPENMP
+#define CHILD_NCCL_CALL_RANK(errCode, cmd, msg) CHILD_NCCL_CALL_BASE(cmd, msg, OMP_CANCEL_FOR, errCode)
+#define CHILD_NCCL_CALL_NON_BLOCKING_RANK(errCode, msg, localRank) CHILD_NCCL_CALL_NON_BLOCKING_BASE(msg, localRank, OMP_CANCEL_FOR, errCode)
+#else
+#define CHILD_NCCL_CALL_RANK(errCode, cmd, msg) CHILD_NCCL_CALL(cmd, msg)
+#define CHILD_NCCL_CALL_NON_BLOCKING_RANK(errCode, msg, localRank) CHILD_NCCL_CALL_NON_BLOCKING(msg, localRank)
+#endif
+
 namespace RcclUnitTesting
 {
-  TestBedChild::TestBedChild(int const childId, bool const verbose, int const printValues)
+  TestBedChild::TestBedChild(int const childId, bool const verbose, int const printValues, bool const useRankThreading)
   {
     this->childId = childId;
     this->verbose = verbose;
     this->printValues = printValues;
+    this->useRankThreading = useRankThreading;
```

Comment:
If this is only going to be used to determine numThreadsToUse, I'd recommend just storing numThreadsToUse here instead.

Reply:
`numThreadsToUse` gets set to `numRanksToExecute`, which isn't known until it is read from the pipe.

Reply:
Ah - right!